Added NER notebook using pretrained transformer model and wikigold da…

kehuangms opened the pull request:
### Description
<!--- Describe your changes in detail -->
<!--- Why is this change required? What problem does it solve? -->
Added NER notebook using pretrained transformer model and wikigold dataset.

### Related Issues
<!--- If it fixes an open issue, please link to the issue here. -->


### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project, as detailed in our [contribution guidelines](../CONTRIBUTING.md).
- [ ] I have added tests.
- [ ] I have updated the documentation accordingly.

saidbleik commented on examples/named_entity_recognition/ner_wikigold_transformer.ipynb:
```diff
@@ -0,0 +1,318 @@
+    "import sys\n",
+    "import os\n",
+    "\n",
+    "nlp_path = os.path.abspath('../../')\n",
```
we don't need this anymore, if utils_nlp is installed as a pip package

saidbleik commented on examples/named_entity_recognition/ner_wikigold_transformer.ipynb:
```diff
@@ -0,0 +1,318 @@
+    "    NUM_TRAIN_EPOCHS = 1\n",
+    "\n",
+    "# the data path used to save the downloaded data file\n",
+    "DATA_PATH = \"./temp\"\n",
```
We're replacing these temp paths with TemporaryDirectory
https://github.com/microsoft/nlp/blob/ee67aa0108a6e1928a068c66227242f94ef8c3a4/examples/text_classification/tc_mnli_transformers.ipynb

hlums commented on examples/named_entity_recognition/ner_wikigold_transformer.ipynb:
```diff
@@ -0,0 +1,318 @@
+{
```
<p>DO_LOWER_CASE should be True when using an uncased model. </p><p>Any reason we want to fix NUMBER_GPU? If it's not set, our utility function will automatically detect and use all available ones. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/nlp/pull/449/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

kehuangms replied:
<p>Changed the DO_LOWER_CASE = True. </p><p> </p><p>It seems that the default value for NUMBER_GPU in different functions are inconsistent. For example, previous this could be set to "None" to use all GPUs. But recently the default value is "1" and it cannot be set to "None", otherwise there will be an error. I was using CPU to test the code and I have to set it as an integer even it's not necessary. I have changed the default values in NER python file so this will be hidden. </p>

hlums commented on examples/named_entity_recognition/ner_wikigold_transformer.ipynb:
```diff
@@ -0,0 +1,318 @@
+{
```
<p>Can we rename the maxLen arg to max_len to follow the same variable naming convention?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/nlp/pull/449/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

kehuangms replied:
<p>Changed. I just cannot control to use hump naming style. </p>

hlums commented on examples/named_entity_recognition/ner_wikigold_transformer.ipynb:
```diff
@@ -0,0 +1,318 @@
+{
```
<p>Would be nice to show the evaluation results. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/nlp/pull/449/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

kehuangms replied:
<p>Added scrapbook.</p>

kehuangms commented:
> I think it would be good to have the tests

I just replaced the original bert notebook with the new transform notebook, and updated the integration test. The unit test part does not need to be do any change since I did not modify the original functions. A few new functions are added and it's an integration of all original functions so no need to add more unit tests.
